File: src/main/java/keeyeong/codility/TimeComplexity/TapeEquilibrium.java
package keeyeong.codility.TimeComplexity;

public class TapeEquilibrium {
	public int solution(int[] A) {
		int left = A[0];
		int right = 0;
		int diff;
		for (int i = 1; i < A.length; i++) {
			right += A[i];
		}
		int min = Math.abs(left - right);
		for (int i = 1; i < A.length - 1; i++) {
			left += A[i];
			right -= A[i];
			diff = Math.abs(left - right);
			if (diff < min) {
				min = diff;
			}
		}
		return min;
	}
}

File: src/main/python/TimeComplexity/TapeEquilibrium.py
def solution(A):
	left = A[0]
	right = sum(A[1:])
	min = abs(left - right)
	# print "left=%d, right=%d, diff=%d" % (left, right, min)
	for i in range(1, len(A) - 1):
		left = left + A[i]
		right = right - A[i]
		diff = abs(left - right)
		if diff < min:
			min = diff
		# print "left=%d, right=%d, diff=%d" % (left, right, diff)
	return min
